### DISCUSSION TITLE
Unable to auto-detect the network interface.

### DISCUSSION THREAD

**Discussion by yangshubin27400430** [gopro/OpenGoPro#98] [Q&A] [answered]
I downloaded the latest code.
Bluetooth demo can be connected to my GoPro10
[OpenGoPro-main\demos\python\tutorial\tutorial_modules\tutorial_1_connect_ble\ble_connect.py]

But I had an error trying the WiFi module
 [OpenGoPro-main\demos\python\tutorial\tutorial_modules\tutorial_5_connect_wifi\wifi_enable_and_connect.py]
error1 ::"open_gopro.wifi_controller" could not be resolved
So I changed the code
   from open_gopro.wifi_controller import Wireless  --->>  from open_gopro.wifi.adapters import Wireless

But there are new mistakes
error2 ::"Unable to auto-detect the network interface. "

my PC is windows 10
Python is 3.8.1

I don't know what the problem is
thank you

**Comment by dbstpfk** [comment]
Have you ever solved it? I am having the same problem.

**Comment by tcamise-gpsw** [comment]
Hello. I think the tutorials might be out of sync with the latest Python SDK updates. I'll look into this.

Some other notes:
- the Wifi controller is parsing your systems responses and thus far is only configured to handle english. So if you are using any language other than english, it will not work
- you can test this outside of the tutorials (assuming you have installed the OpenGoPro python SDK) via `gopro-wifi`. It should automatically connect to the first GoPro it discovers. If this works, then it is definitely just a problem in the tutorials.

**Comment by tcamise-gpsw** [comment]
FYI I've created an issue to track this: https://github.com/gopro/OpenGoPro/issues/103

**Comment by tcamise-gpsw** [comment, marked as answer]
This should be fixed now via #109. I hard-coded the OpenGoPro version to prevent this from happening again as well as added a test suite.
